### PR TITLE
Sandy Thomson Chapter 8 Answers

### PR DESCRIPTION
**Pull request by SandyThomson**

### CODE REVIEW

File: answers/chapter8.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  
+  3. Packages are stored in file system directories.
+  
+  4. protected is less restrictive than default, as it allows all subclasses to access a member, even if the subclass isn't in the same package.
+  
+  5. If a member is public, it can be used directly by any class. If a member is protected, it can be used by any class that _extends_ the class containing the member.
+  
+  6. Erm, the feature of defining an interface, and then implementing it many times. Is this a trick question?
+  
+  7. In theory, no limit to either. However in reality, a class can't implement more than 65535 interfaces (source: SO)
```
Comment (SimonGross) [inline answers/chapter8.md:15]: :+1: for research. The book doesn't specify a limit, but I guess they were talking in theory.

File: answers/chapter8.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  2. A namespace is essentially a way of partitioning Java programs for access control and to avoid naming conflicts / collisions. 
+  This is managed by the package keyword, and classes in a package are kept in a directory structure with the same hierarchy.
+  
+  3. Packages are stored in file system directories.
+  
+  4. protected is less restrictive than default, as it allows all subclasses to access a member, even if the subclass isn't in the same package.
+  
+  5. If a member is public, it can be used directly by any class. If a member is protected, it can be used by any class that _extends_ the class containing the member.
+  
+  6. Erm, the feature of defining an interface, and then implementing it many times. Is this a trick question?
```
Comment (SimonGross) [inline answers/chapter8.md:13]: Not a trick question, they really do want you to say "'One interface, multiple methods' is exemplified by interfaces"

File: answers/chapter8.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  1. See IQDemo.java in the default package.
+  
+  2. A namespace is essentially a way of partitioning Java programs for access control and to avoid naming conflicts / collisions. 
+  This is managed by the package keyword, and classes in a package are kept in a directory structure with the same hierarchy.
+  
+  3. Packages are stored in file system directories.
+  
+  4. protected is less restrictive than default, as it allows all subclasses to access a member, even if the subclass isn't in the same package.
+  
+  5. If a member is public, it can be used directly by any class. If a member is protected, it can be used by any class that _extends_ the class containing the member.
```
Comment (SimonGross) [inline answers/chapter8.md:11]: You're correct, but what they were looking for was how you can see these members in the first place. Either by importing, or by direct referencing. For example:

``` java
int answer = package.Class.getAnswer();

import package.Class;
int answer = Class.getAnswer();
```

File: src/IQDemo.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+		qpack.DynQueue dq = new qpack.DynQueue(22); // An example with no
+													// import, for fits and
+													// giggles.
+
+		cq.put('a');
+		fq.put('b');
+		dq.put('c');
+
+		// etc
```
Comment (SimonGross) [inline src/IQDemo.java:19]: Can't really blame you for skipping the rest, the only point of this question was display proper use of imports. I guess its punishment for skipping over the TryThis earlier in the chapter.

Comment (alexdeas) [inline src/IQDemo.java:19]: Kept all the queue implementation classes though.

File: answers/chapter8.md (hunk 2 above)
Comment (alexdeas) [inline answers/chapter8.md:13]: You got tricked into learning Java.

File: answers/chapter8.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+
+  1. See IQDemo.java in the default package.
+  
+  2. A namespace is essentially a way of partitioning Java programs for access control and to avoid naming conflicts / collisions. 
+  This is managed by the package keyword, and classes in a package are kept in a directory structure with the same hierarchy.
```
Comment (alexdeas) [inline answers/chapter8.md:5]: Extra details, no extra marks though

File: src/vehicle/IVehicle.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package vehicle;
+
+// Why not VehicleInterface??? WHy Java book, WHY?
```
Comment (alexdeas) [inline src/vehicle/IVehicle.java:3]: They did it to hurt you.

File: src/vehicle/IVehicle.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
+package vehicle;
+
+// Why not VehicleInterface??? WHy Java book, WHY?
+public interface IVehicle {
+
+	default int range() {
```
Comment (alexdeas) [inline src/vehicle/IVehicle.java:6]: Fancy defaults.

File: answers/chapter8.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  
+  7. In theory, no limit to either. However in reality, a class can't implement more than 65535 interfaces (source: SO)
+  
+  8. Interfaces can be extended by other interfaces
+  
+  9. See IVehicle.java
+  
+  10. Yes. They are constants, and have global access.
+  
+  11. A package is a container for classes. It can also contain interfaces etc.
```
Comment (alexdeas) [inline answers/chapter8.md:23]: Yeah, but true or false?

**Pull request closed without merging**